exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ops/InListOps.kt

```diff
@@ -123,3 +126,70 @@ class TripleInListOp<T1, T2, T3>(
         append(")")
     }
 }
+
+/**
+ * Represents an SQL operator that checks if all columns of a `List` [expr] match any of the lists of
+ * values from [list].
+ *
+ * To inverse the operator and check if the `List` of columns is **not** in [list], set [isInList] to `false`.
+ */
+class MultipleInListOp(
+    override val expr: List<Column<*>>,
+    list: Iterable<List<*>>,
+    isInList: Boolean = true
+) : InListOrNotInListBaseOp<List<*>>(expr, list, isInList) {
+    override val columnTypes: List<Column<*>> = expr
+
+    override fun QueryBuilder.registerValues(values: List<*>) {
+        append("(")
+        expr.forEachIndexed { i, expression ->
+            registerArgument(expression.columnType, values[i])
+            if (i != values.lastIndex) append(", ")
+        }
+        append(")")
+    }
+
+    override fun toQueryBuilder(queryBuilder: QueryBuilder) {
+        // SQL Server does not support IN operator with tuples (or any more than 1 expression on the left-hand side)
+        if (currentDialectIfAvailable !is SQLServerDialect) {
+            super.toQueryBuilder(queryBuilder)
+        } else {
+            queryBuilder {
+                val iterator = list.iterator()
+                if (!iterator.hasNext()) {
+                    if (isInList) {
+                        +FALSE
+                    } else {
+                        +TRUE
+                    }
+                } else {
+                    // Built-in exists(AbstractQuery) cannot be used because row value constructors are not supported
+                    // Alternative: Compound AND + OR operators for each value list ->
+                    // (col_1 = ? AND col_2 = ? AND col_3 = ?) OR (col_1 = ? AND col_2 = ? AND col_3 = ?) OR ... (...)
+                    val transaction = TransactionManager.current()
+                    val columnNames = columnTypes.map { transaction.identity(it) }
+                    val firstValue = iterator.next()
+
+                    when {
+                        isInList -> append("EXISTS (")
+                        else -> append("NOT EXISTS (")
+                    }
+                    append("SELECT * FROM (VALUES ")
```

Comment:
I don't expect that it's possible to do that in a good way here, but probably there is a way to reuse another query builder here.

As I can see the most tricky part is to get `FROM (VALUES (0, 0, 0, 0), (1, 1, 1, 1), (2, 2, 2, 2)) v(num_1, num_2, num_3, num_4)` where there is a list of values and alias with column names.

But if we imagine that we have such a structure, we could use normal `selectAll`, `where` to it without applying low-level query builder operations.

We also have `Exists`,`NotExists` operators.

Reply:
As I mentioned in the comments above these lines, the built-in `exists(AbstractQuery)` operator cannot be used because row value constructors (`SELECT * FROM (VALUES (...), (...))`) are not currently supported by Exposed.

Yes, it would be great to avoid this entirely and reuse an existing query builder, but the only option to do so, as mentioned above, would be to use compound `AND` + `OR` operators:
```kt
// replace the entire else block with this
val valueEqualityOps = mutableListOf<Op<Boolean>>()
val eqOp = if (isInList) ::EqOp else ::NeqOp

iterator.forEach { value ->
    val valueEqualityOp = Op.build {
        expr.zip(value).map { (column, value) ->
            Op.build { eqOp(column, column.wrap(value)) }
        }.compoundAnd()
    }
    valueEqualityOps.add(if (isInList) valueEqualityOp else not(valueEqualityOp))
}

valueEqualityOps.compoundOr().toQueryBuilder(queryBuilder)
```

I think that using the current way is better for the user, more efficient, and with significantly less verbose SQL, but that may be a wrong opinion, so switching to the above to reduce string building would be no problem.